### PR TITLE
Nerfs Autocloning Pt. 2: Autoscan

### PR DESCRIPTION
**Pull request by YakumoChen**
## About The Pull Request
Changes the scan function of autoprocess; All living occupants (pre-death scanning) must be scanned manually, only corpses will automatically scan into the cloning database (and subsequently clone).

## Why It's Good For The Game

Cloning will now require someone press the "Scan" button whenever anyone alive is in the DNA scanner. This will things slightly more tedious for anyone attempting to pre-scan, but will also prevent a lot of cloning cheese available when autocloning becomes possible.

Corpses are immune to this changed behavior. They will still scan as soon as they are shoved into the scanning machine, assuming auto-process is enabled.

**This PR is NOT removing the following:
-Autocloning
-Pre-death scanning for the purposes of autocloning**

**So long as you have a buddy to scan you

What this WILL remove:
-Making autocloners in space/maint where nobody can find them
-Scanning yourself for autoclone immediately after cloning
-Autocloning removing all interaction from medical from the cloner.

## Changelog
:cl: Yakumo Chen
del: Changed cloner autoscanning
balance: Scanning living people now requires someone to operate the cloning computer regardless of part level. (this does not prevent autocloning)
/:cl:

### CODE REVIEW

**Comment by lolman360** [comment]
antag nerf

**Comment by silicons** [comment]
i'm alright with this

**Comment by RogueSteampunker** [comment]
I mean, it stomps antags and non-antags alike who try to make personal cloning booths. Seems like a good change to me.

**Comment by Trilbyspaceclone** [comment]
Why stop dead people auto scanning?
It tends to help when they miss the first scan promp and medical being braindead to auto bag them

**Comment by Sishen1542** [comment]
i agree with trilby, could just make it auto only for corpses

**Comment by davidchan13** [comment]
Prescanning is the only problem with autoclone. As repeatedly stated. Get rid of that and suddenly autoprocess isn't a problem and death is weight if your body isn't recoverable.

**Comment by YakumoChen** [comment]
@Trilbyspaceclone I don't have the coding prowess to make that sort of change. Scanning corpses is not that time-consuming anyway.

I also do not plan for the Autocloning argument to become a thing in this PR. People are very divisive about it, this PR is solving a related issue that sidesteps the argument. If you wanna make a PR removing pre-scanning, go ahead. I will not be reading any posts that relate to autocloning or pre-scanning however.

**Review by silicons — COMMENTED** [review]
@YakumoChen 

could add 
`if(ismob(scanner.occupant))
    var/mob/M = scanner.occpuant
    if(M.stat == DEAD)
        scan_occupant(scanner.occupant)`

up to you though

**Comment by silicons** [comment]
the tabs/indents are fucked up int he above but it should be readable

**Comment by YakumoChen** [comment]
I've done the change with dead people, although I don't agree with it

**Comment by YakumoChen** [comment]
I think trying to make that change broke it, fuck

File: code/game/machinery/computer/cloning.dm [hunk 1]
```diff
@@ -64,7 +64,7 @@
 	if(!(scanner && LAZYLEN(pods) && autoprocess))
 		return
 
-	if(scanner.occupant && scanner.scan_level > 2)
+	if(ismob(scanner.occupant)) var/mob/M = scanner.occpuant if(M.stat == DEAD) //scans dead people automatically, not living.
```
Comment (silicons) [inline code/game/machinery/computer/cloning.dm]: put the var/mob/M = scanner.occupant on the next line under the if(ismob block
then put if(M.stat == DEAD) //comment after that, and shift scan_occupant to be under that block =p

**Comment by nik707** [comment]
@YakumoChen can y ou uhh do the thing

**Comment by Tupinambis** [comment]
This doesn't really do anything to fix the problems with cloning, and just pisses dead people off when doctors/geneticists inevitably never check the cloner. I am strongly against this change. 

I'd rather backup scans be removed entirely, or alternatively, have a delay in which the brain must be intact for the entire duration before backups will work. (Gives time for antags to destroy the brain for clone prevention without fucking over say, miners)

**Comment by YakumoChen** [comment]
Did you even read my pr?


As for commenters who matter, ivr been busy being addicted to star sector, I'll fix it soon

**Comment by YakumoChen** [comment]
good news, it didn't work

**Comment by YakumoChen** [comment]
@kevinz000 help

**Comment by YakumoChen** [comment]
tumbling down

**Pull request closed without merging**